ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -316,82 +315,69 @@ export class AccountCreation extends React.Component<AccountCreationProps, Accou
+      familyName: presenceCheck,
       contactEmail: {
-        presence: {allowEmpty: false},
+        presence: presenceCheck,
         format: {
           pattern: /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/,
           message: 'Invalid email address'
         }
       },
-      streetAddress1: {
-        presence: {allowEmpty: false}
-      },
-      city: {
-        presence: {allowEmpty: false}
-      },
-      state: {
-        presence: {allowEmpty: false}
-      },
-      zipcode: {
-        presence: {allowEmpty: false},
-        format: {
-          pattern: /^[0-9]*$/,
-        }
-      },
-      country: {
-        presence: {allowEmpty: false}
-      },
-      affiliation: {
-        presence: {allowEmpty: this.state.showInstitution}
-      },
-      affiliationRole: {
-        presence: {allowEmpty: !this.state.affiliation}
-      },
-      institutionName: {
-        presence: {allowEmpty: !this.state.showInstitution}
-      },
-      institutionRole: {
-        presence: {allowEmpty: !this.state.showInstitution}
+      streetAddress1: presenceCheck,
+      city: presenceCheck,
+      state: presenceCheck,
+      country: presenceCheck
+    };
+
+    showInstitution ? validationCheck['institution'] = presenceCheck :
+      validationCheck['affiliation'] = presenceCheck;
+
+    if (showInstitution || affiliation !== AffiliationRole.COMMUNITYSCIENTIST) {
+      validationCheck['role'] = presenceCheck;
+    }
+
+
+    const errors = validate({givenName, familyName, contactEmail, streetAddress1, city, state, country,
+      institution, affiliation, role}, validationCheck);
+    this.setState({errors: errors}, () => {
+      if (!this.state.errors) {
+        this.props.setProfile(this.state.profile, 'accountCreationSurvey');
       }
     });
-    this.setState({errors: errors});
   }
 
 
   render() {
     const {
       profile: {
         givenName, familyName, currentPosition, organization,
-        contactEmail, username, areaOfResearch
-      },
-      address: {
-        streetAddress1, streetAddress2, city, state, zipcode, country
+        contactEmail, username, areaOfResearch,
+        address: {
+          streetAddress1, streetAddress2, city, state, zipCode, country
+        },
+        institutionalAffiliations
       },
-      institutionName, institutionRole, showNext, affiliation, affiliationRole, affiliationOther
     } = this.state;
     return <div id='account-creation'
-                style={{'paddingTop': environment.enableAccountPages ? '1.5rem' :
-                      '3rem', 'paddingRight': '3rem', 'paddingLeft': '3rem'}}>
+                style={{paddingTop: environment.enableAccountPages ? '1.5rem' :
+                      '3rem', paddingRight: '3rem', paddingLeft: '3rem'}}>
       <Header>Create your account</Header>
-      {environment.enableAccountPages && !showNext && <div style={{marginTop: '0.5rem'}}>
+      {environment.enableAccountPages && <div style={{marginTop: '0.5rem'}}>
```

Comment:
What does removing `showNext` do?

Reply:
earlier showNext was controlling the display of next page 'Account creation survey'. However i reverted it back to use the current way routing of login page happens ('sign-in.tsx method 'setProfile')